### DISCUSSION TITLE
Is Test Drive inconsistent?

### DISCUSSION THREAD

**Discussion by ComradeWave** [AvaloniaUI/avalonia-docs#447] [Q&A] [answered]
Hi there, i just started out Avalonia, but after trying the Test Drive, i noticed that it was inconsistent with the usage of MainView.axaml and MainWindow.axaml. In the end you can't complete the exercise if you code in MainWindow (You get an error on the Window tag, and you can't fix it using StackPanels). You can finish the exercise only if you write everything in MainView. Was i dumb or something is wrong?

**Comment by timunie** [comment, marked as answer]
@ComradeWave it depends on the template and IDE used to create a new project. So in case you see MainView, you can just use that instead of MainWindow. Basically MainWindow just presents MainView. For beginners, please also explore Avalonia.Samples: https://github.com/AvaloniaUI/Avalonia.Samples

**Reply by ComradeWave** [reply]
I didn't quite get the difference between MainView and MainWindow, what's different?

**Reply by timunie** [reply]
Some platforms can't have a Window, for example Mobile and Browser. There MainView is shown directly. On the other hand Desktops require a Window. So in order to make all platfroms happy, a MainView is created where all the content goes. MainWindow does show it on Desktop. Basically it's a limitation / requirement of our x-plat support. If you only target Desktop, MainWindow is okay to use directly.

**Reply by ComradeWave** [reply]
I see thanks!
